luskan/src/main/java/org/opencadc/luskan/ws/StorageTapService.java

```diff
@@ -148,38 +155,85 @@ public AvailabilityStatus getStatus() {
+            String url = getServiceUrl(Standards.CRED_PROXY_10, Standards.VOSI_AVAILABILITY, AuthMethod.ANON);
             CheckResource checkResource = new CheckWebService(url);
             //checkResource.check();
 
-            URI usersURI = localAuthority.getServiceURI(Standards.UMS_USERS_01.toString());
-            url = reg.getServiceURL(usersURI, Standards.VOSI_AVAILABILITY, AuthMethod.ANON).toExternalForm();
+            url = getServiceUrl(Standards.UMS_USERS_01, Standards.VOSI_AVAILABILITY, AuthMethod.ANON);
             checkResource = new CheckWebService(url);
             //checkResource.check();
 
-            URI groupsURI = localAuthority.getServiceURI(Standards.GMS_SEARCH_01.toString());
-            url = reg.getServiceURL(groupsURI, Standards.VOSI_AVAILABILITY, AuthMethod.ANON).toExternalForm();
+            url = getServiceUrl(Standards.GMS_SEARCH_01, Standards.VOSI_AVAILABILITY, AuthMethod.ANON);
             checkResource = new CheckWebService(url);
             checkResource.check();
 
         } catch (CheckException ce) {
             // tests determined that the resource is not working
             isGood = false;
             note = ce.getMessage();
-        } catch (Throwable t) {
+        } catch (Throwable throwable) {
             // the test itself failed
-            log.error("test failed", t);
+            StringBuilder sb = new StringBuilder();
+            sb.append("test failed because: ");
+            while (throwable != null) {
+                sb.append(throwable.getMessage()).append(": ");
+                throwable = throwable.getCause();
+            }
+            log.error(sb.toString());
             isGood = false;
-            note = "test failed, reason: " + t;
+            note = sb.toString();
         }
         return new AvailabilityStatus(isGood, null, null, null, note);
     }
 
+    private String getServiceUrl(URI standardID, URI standard, AuthMethod authMethod)
+        throws ResourceNotFoundException {
+        LocalAuthority localAuthority = new LocalAuthority();
+        RegistryClient reg = new RegistryClient();
+
+        URI resourceID;
+        try {
+            resourceID = localAuthority.getServiceURI(standardID.toString());
+        } catch (NoSuchElementException e) {
+            String message = String.format("service URI not found in LocalAuthority for standardID %s", standardID);
+            throw new ResourceNotFoundException(message);
+        }
+
+        Capabilities capabilities;
+        try {
+            capabilities = reg.getCapabilities(resourceID);
+        } catch (IOException e) {
+            String message = String.format("io error getting capabilities for resourceID %s, standardID %s, because %s",
+                                           resourceID, standardID, e.getMessage());
+            throw new ResourceNotFoundException(message);
+        }
+
```

Comment:
From here down, `ResourceNotFoundException` is not the correct error if a service does not implement the standard you are interested in nor if it does not support a specific AuthMethod (the latter especially is not a resource in any sense). 

Yes, this is captured above and not thrown as it would be in a library, but things like this tend to be copied/moved into libraries where choice of exception becomes entrenched.

Probably `UnsupportedOperationException("service %s does not provide %s", ...)` would work for findCapability. I don't know about findInterface(AuthMethod)... it's not an operation and is never going to happen in practice so I'd probably just go with `RuntimeException("unexpected: service %s capability %s does not support auth: %s", ...)`

Reply:
Updated the error messages, thanks, much better.